### PR TITLE
Add infection, add unit tests, add GitHub action and keep libs up to date

### PR DESCRIPTION
**Pull request by milanowicz**
- Change minimum PHP Version from 7.3 to 7.4
- Add Infection
- Add Unit Tests
- Add Psalm
- Add GitHub Actions to test for PHP env
- Keep libs up to date
- Format code to PSR-12
- Remove Travis CI

### CODE REVIEW

File: src/GenericCollection.php [hunk 1]
```diff
@@ -111,7 +114,7 @@ public function offsetGet($offset)
     public function offsetSet($offset, $value)
```
Comment (Das-Ente) [inline src/GenericCollection.php]: void

File: src/GenericCollection.php [hunk 2]
```diff
@@ -159,20 +162,20 @@ public function count()
...
+    public function addItemList($itemList): self
     {
         foreach ($itemList as $item) {
             if (!$this->checkType($item)) {
-                throw new \InvalidArgumentException('Invalid type');
+                throw new InvalidArgumentException('Invalid type to add in list');
             }
 
-            $this->itemList[] = $item;
+            $this->itemList[] = clone $item;
```
Comment (Das-Ente) [inline src/GenericCollection.php]: This could result in a PHP Warning when type it is a scalar type.

File: src/GenericCollection.php [hunk 3]
```diff
@@ -333,7 +335,7 @@ public function partition(Closure $func)
      */
     public function reduce(Closure $func)
```
Comment (Das-Ente) [inline src/GenericCollection.php:336]: :array

File: src/GenericCollection.php (hunk 2 above)
Comment (milanowicz) [reply src/GenericCollection.php]: Done

File: src/GenericCollection.php (hunk 1 above)
Comment (milanowicz) [reply src/GenericCollection.php]: Done

File: src/GenericCollection.php (hunk 3 above)
Comment (milanowicz) [reply src/GenericCollection.php:336]: I tried it, but it doesn't work. Return is mixed from PHP and not array.

https://www.php.net/manual/en/function.array-reduce.php

Comment (milanowicz) [reply src/GenericCollection.php:336]: Edit in PR for PHP 8.x:
https://github.com/jtl-software/php-generic-collection/pull/9

**Pull request merged**